pandas/core/reshape/reshape.py

```diff
@@ -1033,22 +1037,26 @@ def wide_to_long(df, stubnames, i, j, sep="", suffix='\d+'):
     -----
     All extra variables are left untouched. This simply uses
     `pandas.melt` under the hood, but is hard-coded to "do the right thing"
-    in a typicaly case.
+    in a typical case.
     """
     def get_var_names(df, stub, sep, suffix):
-        regex = "^{stub}{sep}{suffix}".format(
-            stub=re.escape(stub), sep=re.escape(sep), suffix=suffix)
-        return df.filter(regex=regex).columns.tolist()
+        regex = '^{0}{1}{2}$'.format(re.escape(stub), re.escape(sep), suffix)
+        return [col for col in df.columns if re.match(regex, col)]
 
     def melt_stub(df, stub, i, j, value_vars, sep):
         newdf = melt(df, id_vars=i, value_vars=value_vars,
                      value_name=stub.rstrip(sep), var_name=j)
         newdf[j] = Categorical(newdf[j])
         newdf[j] = newdf[j].str.replace(re.escape(stub + sep), "")
 
+        # GH17627 Cast numerics suffixes to int/float
+        newdf[j] = newdf[j].astype('int64', errors='ignore')
```

Comment:
just use ``to_numeric(..., errors='ignore')`` and you will get what you need here. This is a soft conversion.

Reply:
Thanks. Changed.